packages/stylist-core/src/bow.rs

```diff
@@ -0,0 +1,62 @@
+use std::hash::{Hash, Hasher};
+use std::ops::Deref;
+use Bow::*;
+
+// Read as "box on write", a copy on write type where the
+// alternative is a Box.
+#[derive(Debug)]
+pub enum Bow<'a, T: 'a + ?Sized> {
```

Comment:
This type is used as an alternative for `Cow` where the `Owned` alternative should be boxed to break cycles, as in `RuleContent`. Should have very similar semantics.